### PR TITLE
added TestOptOutOptIn and moved some code to functions

### PR DESCRIPTION
**Pull request by lina-nikiforova**
Added new test to verify that insights operator opt-in/opt-out works

### CODE REVIEW

File: test/integration/basic_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		t.Fatal(err.Error())
+	}
+
+	// Edit the `.dockerconfigjson` file that was downloaded.
+	// Remove the `cloud.openshift.com` JSON entry.
+	newPullSecret := pullSecret.DeepCopy()
+	secretConfig := newPullSecret.Data[".dockerconfigjson"]
+
+	obj := map[string]interface{}{}
+	json.Unmarshal([]byte(secretConfig), &obj)
```
Comment (tisnik) [inline test/integration/basic_test.go]: I'd probably add a check there - if `json.Unmashal` is ok or not

File: test/integration/basic_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	modifiedConfig, err := json.Marshal(obj)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	newPullSecret.Data[".dockerconfigjson"] = modifiedConfig
+
+	objOld := map[string]interface{}{}
+	json.Unmarshal([]byte(pullSecret.Data[".dockerconfigjson"]), &objOld)
```
Comment (tisnik) [inline test/integration/basic_test.go]: dtto

File: test/integration/basic_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	credsOld := objOld["auths"].(map[string]interface{})
+	fmt.Println(credsOld)
```
Comment (tisnik) [inline test/integration/basic_test.go]: prob. `t.Log(credsOld)` might be better inside unit test

File: test/integration/main_test.go [hunk 4]
```diff
@@ -33,8 +39,94 @@ func KubeClient() (result *kubernetes.Clientset) {
...
+			}
+			if pod.Status.Phase != "Running" {
+				return false, nil
+			}
+		}
+
+		fmt.Println("the pods are created")
+		return true, nil
+	})
+	fmt.Print(errPod)
```
Comment (tisnik) [inline test/integration/main_test.go]: `t.Log`?

File: test/integration/basic_test.go (hunk 3 above)
Comment (lina-nikiforova) [reply test/integration/basic_test.go]: oh, that part needs to be removed, it's not used

File: test/integration/basic_test.go (hunk 2 above)
Comment (lina-nikiforova) [reply test/integration/basic_test.go]: this part is removed now

**Comment by lina-nikiforova** [comment]
ci/prow/e2e-aws failed but shouldn't be related to my change

**Comment by lina-nikiforova** [comment]
e2e-aws is kinda weird, keeps failing with different errors each time

**Comment by iNecas** [comment]
I wonder if there is any downside of using `t.Logf` instead of `fmt.Printf`. If not, I would probably prefer the `t.Log` variant

**Comment by tisnik** [comment]
@iNecas `t.Logf` is better in this case (at least we don't need to play with those `\n` everywhere :)

**Comment by lina-nikiforova** [comment]
@iNecas @tisnik thank you :)

**Comment by lina-nikiforova** [comment]
test failure is not related to tests themselves 
```
level=error msg="Cluster operator monitoring Degraded is True with UpdatingkubeStateMetricsFailed: Failed to rollout the stack. Error: running task Updating kube-state-metrics failed: reconciling kube-state-metrics Deployment failed: updating deployment object failed: waiting for DeploymentRollout of kube-state-metrics: deployment kube-state-metrics is not ready. status: (replicas: 1, updated: 1, ready: 0, unavailable: 1)"
level=info msg="Cluster operator monitoring Available is False with : "
level=info msg="Cluster operator support Disabled is True with NotAuthorized: Reporting was not allowed: your Red Hat account is not enabled for remote support or your token has expired"
level=fatal msg="failed to initialize the cluster: Some cluster operators are still updating: image-registry, ingress, monitoring"
```

**Pull request merged**